### PR TITLE
feat: read installable paths from stdin

### LINKED ISSUE #7437: cli: read "paths" from stdin for commands that take them
**Is your feature request related to a problem? Please describe.**
If the user wants to pass a list of paths to say `nix path-info` or `nix-build`, they run the risk of recieving a "too many args" OS error.

Since passing derivations around can be very efficient when handled in batches it would be nice not to have a cap on this. The current workaround is `xargs`, which I guess is fine, but not exactly intuitive or necessarily efficient.

**Describe the solution you'd like**
I could see it two ways, if the command recieves a non-empty stdin, simply add it to whatever list of paths already exist in the cli, or we could make it explicit and require something like `nix-build -` to specify reading paths from stdin.

**Describe alternatives you've considered**
Basically just keep using xargs, but even if the arg limit is not an issue, shell pipelines are far more readable than nested process substititions, e.g:
```console
$ which svn | nix-store -qR | nix-store --realize
# is nicer than
$ nix-build $(nix-store -qR $(which svn))
```

**Additional context**
Hit this a couple of times when trying to ensure the entire build closure is cached by parsing the results of a `--dry-run`: [divnix/std-action/run/build.sh#14](https://github.com/divnix/std-action/blob/b01efeb31dca2df9f7672233ac1de92068be3671/run/build.sh#L14)

### PR DESCRIPTION
**Pull request by nrdxp**
Resolves #7437 for new `nix` commands and for `nix-store` by adding a new flag for these commands `--stdin`.

I decided to add this functionality to the legacy `nix-store` command for primarily one reason: passing a large amount of paths (more than os arg limit) to `nix-store --export`.

If paths are also passed on the cli they will be combined with the ones from standard input.

### CODE REVIEW

File: doc/manual/src/command-ref/nix-store.md [hunk 1]
```diff
@@ -54,6 +54,11 @@ have an effect.
     created by sequentially numbering symlinks beyond the first one
     (e.g., `foo`, `foo-2`, `foo-3`, and so on).
 
+  - <span id="opt-stdin">[`--stdin`](#opt-stdin)</span>
+
+    Read *paths…* from the standard input. Useful for chaining
+    nix-store commands.
```
Comment (fricklerhandwerk) [inline doc/manual/src/command-ref/nix-store.md:60]:
```suggestion
    Read *paths…* from the standard input.
    Useful for chaining `nix-store` commands.
```

**Comment by nrdxp** [comment]
> Looks good, add to release notes?

Sounds good, but it seems the notes are empty. Is there a standard format these should be in? I can find anything in the git log.

**Comment by tomberek** [comment]
> > Looks good, add to release notes?
> 
> Sounds good, but it seems the notes are empty. Is there a standard format these should be in? I can't find a good example in the git log. Something similar to the nixpkgs release notes I imagine.

We place things into https://github.com/NixOS/nix/blob/master/doc/manual/src/release-notes/rl-next.md

**Comment by nrdxp** [comment]
Apologies for delay, release note was added.

**Comment by blaggacao** [comment]
This is an increasingly common issue for us when scripting large closures.

At least according to GH, this is ready for merging?

**Comment by nrdxp** [comment]
> this is ready for merging?

looks like we have a merge conflict now, but I will rebase and it should be ready from there.

**Review by thufschmitt — APPROVED** [review]
Yup' that's good.

Thanks @nrdxp, that should be merged as soon as the CI is happy

**Comment by tomberek** [comment]
@nrdxp it seems that this means that:
```
true | nix COMMAND --stdin
```
will have the same impact as `nix COMMAND`, which is a bit surprising.

Should we check for the case of zero installables via stdin and error out?

**Pull request merged**